core/app/models/spree/order.rb

```diff
@@ -73,6 +73,11 @@ class CannotRebuildShipments < StandardError; end
     has_many :variants, through: :line_items
     has_many :products, through: :variants
 
+    if Spree.solidus_gem_version > Gem::Version.new('2.5.0')
+      has_many :order_stock_locations, class_name: 'Spree::OrderStockLocation'
+      has_many :stock_locations, through: :order_stock_locations
+    end
```

Comment:
What about adding this code to the [legacy stock system](https://github.com/solidusio-contrib/solidus_legacy_stock_system) extension instead?

Reply:
☝️ This seems like a lower touch fix.

Reply:
@kennyadsl @jacobherrington I've been trying to implement this extension-level, but so far it is not working :-( I'll let you know if I come up with a more clean solution